### PR TITLE
fix(filters): Editable number range filter should be unique

### PR DESCRIPTION
**Pull request by annacv**
[EMP-1228](https://searchbroker.atlassian.net/browse/EMP-1228)

## Motivation and context
Make this filter unique, so only one range is applied if we set some ranges consecutively.

- [ ] Dependencies. If any, specify:
- [x] Open issue. If applicable, link:

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [x] Change requires a documentation update

## What is the destination branch of this PR?
- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?
- Setting a range and editing it: it should only show one filter as selected.
- Setting a range and also selecting other filters: clicking the clear filters button should remove them all from the selected filters list.
- Setting a range and deselecting it by clicking on it: the filter should be removed from the selected filters list.

# Test with Tous endpoint
1. Remove the built-in `salePrice` facet from the `aside.vue` script
2. Replace the snippet config in `base-config.ts` with:
  ```
    instance: 'tous',
    scope: 'desktop',
    env: 'staging',
    lang: 'es',
    uiLang: 'es',
    currency: 'EUR',
    consent: true,
    documentDirection: 'ltr',
    store: 'es',
  ```
4. Ad this override to the `adapter.ts`:
```
const getFacetConfigWithEditable = (facetId: string, type: PlatformFacetType): FacetConfig =>
{
  if(facetId === 'salePrice') {
    let salePriceConfig = getFacetConfig('editable-range');
    const getSalePriceFilterId = (context?: MapperContext) => (context?.requestParameters?.filter as string[])?.find(filter => filter.includes('salePrice'));

    salePriceConfig.schema = {
      ...salePriceConfig.schema,
      id: ({filter}, $context) => {
        const currentSalePriceFilterId = getSalePriceFilterId($context);

        return currentSalePriceFilterId ?? filter;
      },
      range: ({value}, $context) => {
        const currentSalePriceFilterId = getSalePriceFilterId($context);

        let stringRage = currentSalePriceFilterId?.split(':')[1] ?? value;

        const min = Number(stringRage.split('-')[0]);
        const max = Number(stringRage.split('-')[1]);

        return {
          min: Number.isNaN(min) ? null : min,
          max: Number.isNaN(max) ? null : max
        }
      }
    }

    return salePriceConfig;
  }

  return getFacetConfig(type);
}

facetSchema.$override<PlatformFacet, Partial<NumberRangeFacet>>({
  modelName: ({ type, facet }) => getFacetConfigWithEditable(facet, type).modelName as any,
  filters: {
    $path: 'values',
    $subSchema: ({ type, facet }) => getFacetConfigWithEditable(facet, type).schema,
    $context: {
      test: 'test'
    }
  }
});
```
5. Test the facet

[EMP-1228]: https://searchbroker.atlassian.net/browse/EMP-1228?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/x-components/src/views/home/aside.vue [hunk 1]
```diff
@@ -6,7 +6,7 @@
       <template #default="{ filter }">
         <SimpleFilter :filter="filter" class="x-facet-filter-success">
           <template #label>
-            {{ filter.label ?? `${filter.range.min} - ${filter.range.max}` }}
+            {{ filter.label ?? filter.id }}
```
Comment (annacv) [inline packages/x-components/src/views/home/aside.vue:9]: The id is set as an alternative label. If the min/max range were used, when the page is reloaded or we perform another search, the filter's getter won't keep that values as the filter model changes to a RawFilter.

**Review by mavmaf — CHANGES_REQUESTED** [review]
Please take a look to the aside menu: the range is added to filters which I believe don't belongs to (brand one, for instance). 
Also, with current code, simple-filters E2E fails 🤔 
image

**Review by CachedaCodes — CHANGES_REQUESTED** [review]
gj!

**Review by lauramargar — DISMISSED** [review]
🎉

**Review by lauramargar — APPROVED** [review]

**Review by mavmaf — APPROVED** [review]

**Review by annacv — COMMENTED** [review]
Hi! Good job!!!

I have been testing and found a use case where the `RangeFilter` behaves differently than other filters when we reload the page. This could be held in another task: 

- Test 1 --> [Hierarchical Filter Test](http://localhost:8080/?filter=promotion%3Ablack%20friday&q=bolso&scroll=195890681): Select a Hierarchical Filter and reload the page, filter is kept in the URL and the results view is rendered with the filter selected
- Test 2 --> [Number Range Filter Test](http://localhost:8080/?filter=price%3A0-10&q=bolso): Select a Number Range Filter and reload the page, filter is kept in the URL, but the results view is not loaded
- Test 3 --> [Editable Number Range Filter Test](http://localhost:8080/?filter=salePrice%3A1-200&q=collar): Select an Editable Number Range Filter and reload the page, filter is kept in the URL and the results view is kept with the filter selected
- Test 4 --> [Hierarchical + Editable Number Range test](http://localhost:8080/?filter=promotion%3Arebajas&filter=salePrice%3A2-200&q=collar): Select Select a Hierarchical Filter and an Editable Number Range Filter and reload the page, filters are kept in the URL and the results view is rendered with the filters selected
- Test 5 --> [Several filter types test](http://localhost:8080/?filter=promotion%3Arebajas&filter=price%3A0-10&filter=salePrice%3A2-200&q=collar): Select Select a Hierarchical Filter, a Number Range Filter, and an Editable Number Range Filter and reload the page, filters are kept in the URL, but the results view is not loaded

**Pull request merged**